### ISSUE TITLE
.NET5 Support

### ISSUE THREAD

**Issue by Samuel12321** [naudio/NAudio#711]

**Comment by forghan89** [comment]
And also update to netstandard2.1


https://devblogs.microsoft.com/dotnet/announcing-net-standard-2-1/
https://github.com/dotnet/standard/blob/master/docs/planning/netstandard-2.1/README.md

**Comment by jwosty** [comment]
If support for .net standard 2.1 were to be included, the .net standard 2.0 target should not be removed due to .Net Framework only going up to standard 2.0.

**Comment by jwosty** [comment]
Is there any particular reason to do this? .Net 5 projects can already consume .Net Core 3 packages just fine. Libraries should also target the lowest versions of their dependencies possible, so that would mean .Net Core 3 and .Net Standard 2.1. Would there be any .Net 5 specific features that NAudio could take advantage of?

**Comment by markheath** [comment]
Yes, there's no obvious reason why NAudio should target netstandard2.1 instead of 2.0. It can be used just fine in .NET 5, and targetting netstandard2.0 makes it available to a much wider audience

**Comment by KvanTTT** [comment]
Also, it could target multiple standards: netstandard2.0 and netstandard2.1

**Comment by jwosty** [comment]
Right, multitargeting should definitely be done if it makes consuming the library easier in certain cases.

I am not familiar with this reference redirection you're talking about @KvanTTT -- what does that mean? (Perhaps it's a good reason for multitargeting)

**Comment by KvanTTT** [comment]
Something like that: https://github.com/antlr/antlr4/issues/2984

**Comment by jwosty** [comment]
On second thought, after #591 is eventually resolved (probably by @markheath's experimental v2 branch; see #593 for more info), it may be useful to add `net5.0-Windows` targets to the relevant libraries to further indicate that they are Windows only. That way, the core NAudio library could keep the platform-less target and you could easily tell what's what.

**Comment by markheath** [comment]
It will only be NAudio.Core that will be cross platform. Everything else is essentially Windows only. (Possibly could look to take the Windows-specific bits of NAudio.Midi out as the message definitions could be useful on their own).

**Comment by markheath** [comment]
Closing this as NAudio 3 has reorganized things a lot to keep all the cross-platform capabilities isolated from the windows-specific features which will hopefully help in the future
